File: estrutural - proxy/Proxy01.py
#Objeto Real

class ServicoReal:
    def executar(self):
        print("executando servico real....")
        

#proxy

class ProxyServico:
    def __init__(self):
        self.servicoReal = ServicoReal()
        
    def executar(self):
        print("Antes da execução do servico....")
        self.servicoReal.executar()
        print("Depois da execução do servico..")
        

#uso do proxy

proxy = ProxyServico()
proxy.executar() 

File: estrutural - proxy/Proxy01.js
//Objeto real
//teste vscode dev

class ServicoReal{
    executar(){
        console.log("servico real executando...")
    }
}


//proxy

class ProxyServico{
    constructor(){
        this.servicoReal = new ServicoReal();
    }

    executar(){
        console.log("Antes de executar o servico ...")
        this.servicoReal.executar();
        console.log("Depois de executar o servico..")
    }
}


//uso do proxy

const proxy = new ProxyServico();

proxy.executar();